arches/app/models/models.py

```diff
@@ -523,20 +523,149 @@ def is_editable(self):
         else:
             return True
 
-    def get_published_graph(self, language=None, raise_if_missing=False):
+    def should_use_published_graph(self):
+        return bool(
+            self.publication
+            and not self.source_identifier
```

Comment:
To avoid eagerly fetching:
```suggestion
            self.publication_id
            and not self.source_identifier_id
```